### PR TITLE
Bump version and update changelog for 0.5.2 release

### PR DESCRIPTION
**Pull request by dbczumar**

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -1,6 +1,16 @@
...
 =========
 
+0.5.2 (2018-08-24)
+------------------
+
+MLflow 0.5.2 is a patch release on top of 0.5.1 containing only bug fixes and no breaking changes or features.
+
+Bug fixes:
+
+- Fix a bug with ECR client creation that caused ``mlflow.sagemaker.deploy()`` to fail when searching for a deployment Docker image.
```
Comment (mateiz) [inline CHANGELOG.rst]: You should reference the PR and author involved here, as we did on the previous release notes

Comment (dbczumar) [reply CHANGELOG.rst]: Done!

**Comment by andrewmchen** [comment]
Should this be merged only to the release branch?

**Comment by dbczumar** [comment]
@andrewmchen That seems reasonable. We'll then merge a version change to `0.5.3dev0` into `master`.

**Review by mateiz — APPROVED** [review]
LGTM

**Pull request merged**